### ISSUE TITLE
Support omitting first and last corner

### ISSUE THREAD

**Issue by fancsali** [dhruvasagar/vim-table-mode#231]
Many MarkDown flavours - and lazy people alike - prefer not to have the `|` characters added at the sides of tables.

In other words, instead of this:

```
| foo | bar | baz |
| 1     |    2  |     3 |
```

one would use:

```
foo | bar | baz
1     |    2  |     3
```

However, at the moment if using the latter format table-mode would not recognise the table - at least not for the `\tr` mapping and `:TableModeRealing` in general.

Can this feature be added perhaps?

**Comment by dhruvasagar** [comment]
@fancsali Yes it can be supported, the table mode abstracts the regular expressions that match the start / end of a table, those would require some tweaking to get this working.
